### DISCUSSION TITLE
Trying to /give or /setblock custo mspawners isnt working. Im aware its a very old essentials issue, wondering

### DISCUSSION THREAD

**Discussion by thomasdaley2** [EssentialsX/Essentials#4139] [Q&A] [answered]
Custom spawners. Trying to /give or /setblock custom spawners isnt working. Im aware its a very old essentials issue, wondering if anything has been done to correct it.

No the answer isnt /minecraft:give, ive seen that and tried it.  Im currently using a test spigot server with only essentials installed 1.16.4

/give businessskeleton minecraft:spawner{BlockEntityTag:{SpawnCount:4,SpawnRange:4,:50,MaxSpawnDelay:500,MaxNearbyEntities:20,RequiredPlayerRange:20,SpawnData:{id:"minecraft:skeleton",NoGravity:0b,CanPickUpLoot:0b},SpawnPotentials:[{Weight:1,Entity:{id:"minecraft:skeleton",NoGravity:0b,CanPickUpLoot:0b}}]}} 1

That code works fine in vanilla, but ass soon as essentials is added.... doesnt work.

**Comment by pop4959** [comment]
Yeah it is indeed an old issue now. There has been some recent discussion of this in https://github.com/EssentialsX/Essentials/issues/3811 and more so in https://github.com/EssentialsX/Essentials/pull/3963 but there isn't a fix we are committed to yet. As I mentioned in the latter post you should still be able to use something like `/i skeletonspawner` to place a skeleton spawner as in your example, but of course the vanilla command no longer works still currently when Essentials is present.

**Reply by thomasdaley2** [reply]
Yeah i have seen this i/ thing  it doesnt allow custom spawners still... trying to make skeles without bows or armor.

**Reply by thomasdaley2** [reply]
I even just removed essentials spawned the custom spawner got it correct... reloaded with essentials and the placed one works but the one in had becomes a pig once placed.

I would say a config option to turn off spawner control would probably fix the issue

**Comment by triagonal** [comment]
I was just able to get this to work fine while having only EssentialsX (+LP, Vault) installed. 

```
[08:51:05 INFO]: Server version: 1.16.5-R0.1-SNAPSHOT git-Paper-597 (MC: 1.16.5)
[08:51:05 INFO]: EssentialsX version: 2.19.0-dev+113-0d640d1
[08:51:05 INFO]: LuckPerms version: 5.3.30
[08:51:05 INFO]: Vault version: 1.7.3-b131
```

I used the command you provided above with `/minecraft:give` (had to remove the `:50` on its own right after the SpawnRange tag since it was invalid), made sure that I didn't have any `essentials.spawnerconvert.` child nodes (I did this by using a default group account, though the same thing could be achieved by setting `essentials.spawnerconvert.*` to false) and it placed the intended spawner fine.

The only issue I ran into was that I was testing on a non-OP account, so I also needed to grant myself `minecraft.place.nbt` for the spawner to be placed correctly.

Here's me and my new best friends having a good time.

![image](https://user-images.githubusercontent.com/10545540/117300791-42574680-aebd-11eb-8740-2df67b827237.png)

**Comment by thomasdaley2** [comment]
*Update*  This worked!  Thankyou.

That :50 was an accident when I pasted it here but I streamlined the code to:

/minecraft:give businessskeleton minecraft:spawner{BlockEntityTag:{SpawnData:{id:"minecraft:skeleton",NoGravity:0b,CanPickUpLoot:0b},SpawnPotentials:[{Weight:1,Entity:{id:"minecraft:skeleton",NoGravity:0b,CanPickUpLoot:0b}}]}} 1

Which does the same thing with default spawner settings.

Thank you

**Reply by thomasdaley2** [reply]
I forwarded this to the server owner and we tested it on the server..... still turning to pigs. lol  But you and I have gotten it working so there must be some other setting somewhere thats interrupting this

**Reply by triagonal** [reply]
AFAIK there are no other EssentialsX features that should mess with this... as long as the user does not have access to `essentials.spawnerconvert.*` then the plugin should ignore them entirely when they place a spawner.

**Reply by thomasdaley2** [reply]
Im thinking its possible that on my server im OP but on the other one im not.  Ill get the op there to try and place it, but other than that im stumped as it obviously CAN work as both of us got it working.

**Comment by JRoy** [comment, marked as answer]
This has now been fixed in a009b2f9b57d08fc4047260505f925e09edd2489
